core/common/src/main/java/alluxio/Configuration.java

```diff
@@ -518,4 +497,49 @@ private void checkUserFileBufferBytes() {
     Preconditions.checkArgument((usrFileBufferBytes & Integer.MAX_VALUE) == usrFileBufferBytes,
         "Invalid \"" + Constants.USER_FILE_BUFFER_BYTES + "\": " + usrFileBufferBytes);
   }
+
+  /**
+   * Factory class to create configuration instances.
+   */
+  @ThreadSafe
+  public static final class Factory {
+
+    /**
+     * @param properties the source configuration
+     * @return a new configuration with only properties copied from source
+     */
+    public static Configuration create(Map<?, ?> properties) {
```

Comment:
done. And also moved from Factory class to object class---I think it is less ideal to have Factory here since we are not really exposing an interface.